tensorflow/contrib/opt/python/training/model_average_device_setter.py

```diff
@@ -0,0 +1,147 @@
+# Contrib 2017.
+# ==============================================================================
+"""Device function for model-average training."""
```

Comment:
No need for two files since people will not use one without the other. Put everything into `model_average_optimizer.py`.

It would be helpful if you squash your commits too.